### ISSUE TITLE
Can not create Azure Backend with TupleAzureBlobStoreBackend

### ISSUE THREAD

**Issue by benoitLebreton-perso** [fivetran/great_expectations#2501]
Hello,

First, I love great expectations I use it with my projects. My coworkers and I are very impressed so thanks a lot.
With _TupleGCSStoreBackend_ I never had problems but now I try to reproduce it on Azure with _TupleAzureBlobStoreBackend_ (which is recent so it is not surprising to have problem with it)

**Describe the bug**
When I use the CLI to create or list the expectations with `$ great_expectations suite list`, I get the following error.
```
Couldn't instantiate class : TupleAzureBlobStoreBackend with config : 
        store_name              expectations_AZ_store
        container               great-expectations-store
        prefix          expectations
        filepath_suffix         .json
 
__init__() got an unexpected keyword argument 'store_name'

```

**Idea**
When I look at the code I see that contrary to _TupleFilesystemStoreBackend_, _TupleS3StoreBackend_ or _TupleGCSStoreBackend_, _TupleAzureBlobStoreBackend_ doesn't expects 'store_name' argument.
But this argument will be given by `class_instance = class_(**config_with_defaults)` in instantiate_class_from_config function. 

Maybe if we just add "store_name" argument in _TupleAzureBlobStoreBackend_ `__init__()` it would solve this issue.

**To Reproduce**

1.  I installed great-expectations==0.13.11 (the latest version in the moment).
2. In a Azure Storage Account, I created a container called "great-expectations-store" 
3. I set _AZURE_STORAGE_CONNECTION_STRING_ env variable with the connexion string of the Azure Storage account.
4. In great_expectations/great_expectations.yml file, I modified the store section to add an Azure-compatible expectation store.
5. I saw the error in the traceback

To set this .yml file, I imitated the setup of _TupleGCSStoreBackend_ but I replaced the "bucket" field with "container" because the `__init__` of  _TupleAzureBlobStoreBackend_ expects "container" instead of "bucket". We have the same issue with validations_store.
```
stores:
  expectations_store:
    class_name: ExpectationsStore
    store_backend:
      class_name: TupleFilesystemStoreBackend
      base_directory: expectations/
  expectations_AZ_store:
    class_name: ExpectationsStore
    store_backend:
      class_name: TupleAzureBlobStoreBackend
      container: great-expectations-store
      prefix: expectations
```

**Environment (please complete the following information):**
 - Operating System: Linux
 - Great Expectations Version: 0.13.11

**Comment by benoitLebreton-perso** [comment]
Hello,
I forked the develop branch (according to your contributing documentation) and I think I have founded how to solve it.
I have just proposed a solution in PR #2505 . I have tested it with my azure account, I works well :)

**Comment by benoitLebreton-perso** [comment]
Hello again,

I forgot a detail in #2505 I wish that I have seen it yesterday, sorry. 
So I opened an other PR #2513 to solve it. We could reopen #2505 if you prefer.

When it is done I think this issue may be closed.

Have a nice day :)

**Comment by eugmandel** [comment]
@benoitLebreton-perso This is our favorite kind of GitHub issue - one that comes with a PR that addresses it!

**Comment by jarandaf** [comment]
I am getting the same issue with `0.13.12` version (had to downgrade because of #2561). Has this been already solved in latest releases?

**Comment by benoitLebreton-perso** [comment]
Yes it is solved in `0.13.14`
